src/Constraint/MultiConstraint.php

```diff
@@ -117,4 +123,84 @@ public function __toString()
 
         return '[' . implode($this->conjunctive ? ' ' : ' || ', $constraints) . ']';
     }
+    /**
+     * @inheritDoc
+     */
+    public function getLowerBound()
+    {
+        $this->extractBounds();
+
+        return $this->lowerBound;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getUpperBound()
+    {
+        $this->extractBounds();
+
+        return $this->upperBound;
+    }
+
+    private function extractBounds()
+    {
+        if (null !== $this->lowerBound) {
+            return;
+        }
+
+        foreach ($this->constraints as $constraint) {
+            if (!$constraint instanceof BoundsProvidingInterface) {
+                continue; // TODO: or exception? Is it better to ignore those or not?
+            }
+
+            $constraintLower = $constraint->getLowerBound();
+            $constraintUpper = $constraint->getUpperBound();
+
+            if (null === $this->lowerBound && null === $this->upperBound) {
+                $this->lowerBound = $constraintLower;
+                $this->upperBound = $constraintUpper;
+                continue;
+            }
+
+            if ($this->versionCompare($constraintLower, $this->lowerBound, $this->isConjunctive() ? '>' : '<')) {
```

Comment:
I think it'd be nicer to actually distinguish the comparison operator here and just do a regular version compare on the versions, rather than the weird version compare / prepare bound magic below?

Reply:
Yeah, let's discuss that below :)